src/main/java/org/apache/mesos/hdfs/util/DnsResolver.java

```diff
@@ -0,0 +1,96 @@
+package org.apache.mesos.hdfs.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.mesos.Protos;
+import org.apache.mesos.SchedulerDriver;
+import org.apache.mesos.hdfs.Scheduler;
+import org.apache.mesos.hdfs.config.SchedulerConf;
+import org.apache.mesos.hdfs.state.PersistentState;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Created by Abhay Agarwal on 3/12/15.
+ */
+public class DnsResolver {
+  public static final Log log = LogFactory.getLog(Scheduler.class);
+
+  private final Scheduler scheduler;
+  private final SchedulerConf conf;
+  private final PersistentState persistentState;
+
+  public DnsResolver(Scheduler scheduler, SchedulerConf conf, PersistentState persistentState) {
+    this.scheduler = scheduler;
+    this.conf = conf;
+    this.persistentState = persistentState;
+  }
+
+  public boolean journalNodesResolvable() {
+    Set<String> hosts = new HashSet<>();
+    if (conf.usingMesosDns()) {
+      for (int i = conf.getJournalNodeCount(); i > 0; i--)
+        hosts.add(HDFSConstants.JOURNAL_NODE_ID + i + "." + conf.getFrameworkName() + "." + conf.getMesosDnsDomain());
+    } else hosts.addAll(persistentState.getJournalNodes().keySet());
+
+    boolean success = true;
+    for (String host : hosts) {
+      log.info("Resolving DNS for " + host);
+      try {
+        InetAddress.getByName(host);
+        log.info("Successfully found " + host);
+      } catch (SecurityException | IOException e) {
+        log.info("Couldn't resolve host " + host);
+        log.info("Sleeping before retrying.");
```

Comment:
Sleeping? Do you mean that the caller (statusUpdate()) will decline the current offer and wait for another offer before trying again to resolve? This log statement belongs up in the caller where the "sleeping" is actually occurring.